tooling/nargo_cli/tests/compile_failure/dup_trait_implementation_5/src/module4.nr

```diff
@@ -0,0 +1,3 @@
+// another module in the crate implements the same trait + struct
+// a Duplicate error should be thrown
+impl crate::module1::MyTrait for crate::module2::MyStruct { }
```

Comment:
I don't think we need `dup_trait_implementation_4` if we have `dup_trait_implementation_5`, the later seems more general. This might be controversial, but I prefer a smaller set of tests that still cover as many cases as possible versus trying to make a new test for every single possible case.

Reply:
On first glance this sounds reasonable, but I can imagine various scenarios where, in a future refactoring (for example, for supporting non-struct types, or could be something completely different), one of the tests could produce a proper error message, while the other could crash the compiler. Imagine e.g. adding trait+type tuples in a hash table local to the module. That could produce a compiler panic. Compilers are complex beasts and anything could happen. What's the benefit of having a smaller set of tests? I don't understand it and I see very little benefit from having less tests.

Reply:
Smaller/fewer tests are beneficial because if you have too many tests they take increasingly longer to run and get increasingly difficult to find tests, sort them, determine if you already have an existing test for a feature, find differences between similar tests, etc. It is a minor point though, I'm not blocking over this.